docs/contributing/authors.md

```diff
@@ -0,0 +1,270 @@
+
+Valid links are enforced automatically through Sphinx's `linkcheck` builder.
+
+[Configuration of the `linkcheck` builder](https://www.sphinx-doc.org/en/master/usage/configuration.html#options-for-the-linkcheck-builder) is in {file}`docs/Makefile` and {file}`docs/conf.py`.
+
+`linkcheck_ignore` supports regular expression syntax.
+
+When authors add a link to their training, it must be a valid public URL without requiring authentication.
+
+If it is not a valid link, or is private or local, then you must exclude it from `linkcheck` by wrapping it in single backticks.
+
+```md
+Visit the URL `http://www.example.com` for an example.
+```
+
+This will render as follows.
+
+> Visit the URL `http://www.example.com` for an example.
+
+If a link has succumbed to bit rot, then try finding the most recently scraped version on the [Internet Archive Wayback Machine](https://web.archive.org/), and update the link.
+
+To validate links, run the following command.
+
+```shell
+make linkcheck
+```
+
+Open `/_build/presentation/output.txt` for a list of broken links.
+
+```{danger}
+Please do not abuse `linkcheck_ignore`.
+
+There is a special place in hell reserved for contributors who do not bother to update bad links, either dead ones or redirects, causing `linkcheck` to fail.
+And there is a doubly punishing place for those who disable `linkcheck` because there are too many bad links.
+
+Please do not be "that person".
+```
+
+
+(authors-syntax-highlighting-label)=
+
+### Syntax highlighting
+
+Sample code blocks perform syntax highlighting through Pygments.
+Authors must use a proper [Pygments lexer](https://pygments.org/docs/lexers/) and not generate warnings.
+
+Avoid adding comments to code snippets, unless you use valid comment syntax for that language.
+JSON does not allow comments.
+
+Do not indicate elided or omitted code with ellipses (`...` or `…`).
+These are almost never valid syntax and will cause syntax highlighting to fail for the code block.
+
+Some lexers are less than perfect.
+If your code block does not highlight well, then consider specifying a less ambitious lexer, such as `text`.
+
+`jsx` has a complex syntax that is difficult to parse.
+We have high hopes for the project [`jsx-lexer`](https://github.com/fcurella/jsx-lexer).
+We include it in our `requirements.txt` file.
+Please contribute to its further development.
+
+The lexers `html+ng2`, `scss`, `http`, `less` are also suboptimal and particular.
+
+To validate code block syntax, run the following command.
+
+```shell
+make html
+```
+
+An [online demo of all lexers that Pygments supports](https://pygments.org/demo/) may be helpful to test out your code blocks and snippets for syntax highlighting.
+
+
+## Synchronize the Browser While Editing
+
+Use gulp to view changes in the browser while editing documentation.
+
+Install the gulp command line utility.
+
+```shell
+npm install --save-dev gulp-cli
+```
+
+Install the gulp project.
+
+```shell
+npm install --save-dev
```

Comment:
just:
`npm install`

Reply:
Do we care whether we install dependencies into a production environment versus development? If so, then we should use `--save-dev`.

I guess it does not matter because we do not deploy any node modules to the production server anyway. I'll fix this.

Reply:
Fixed in c39f9f5d363b50bc007264b6426ecec63b5bb770.